test/visibility_test/visibility_test.go

```diff
@@ -0,0 +1,527 @@
+				})
+			})
+
+			Context("when authentication is basic", func() {
+				BeforeEach(func() {
+					expectedVisibilities = common.Array{postVisibilityRequest, postVisibilityForAllPlatforms}
+				})
+
+				It("returns the visibilities with the credentials' platform id and the visibilities with null platform id if authn is basic", func() {
+					array := ctx.SMWithBasic.GET("/v1/visibilities").
+						Expect().Status(http.StatusOK).JSON().Object().Value("visibilities").Array()
+
+					for _, v := range array.Iter() {
+						obj := v.Object().Raw()
+						delete(obj, "created_at")
+						delete(obj, "updated_at")
+					}
+					array.Contains(expectedVisibilities...)
+
+				})
+			})
+
+		})
+	})
+
+	Describe("POST", func() {
+		Context("With invalid content type", func() {
+			It("returns 415", func() {
+				ctx.SMWithOAuth.POST("/v1/visibilities").
+					WithText("text").
+					Expect().Status(http.StatusUnsupportedMediaType)
+			})
+		})
+
+		Context("With invalid content JSON", func() {
+			It("returns 400", func() {
+				ctx.SMWithOAuth.POST("/v1/visibilities").
+					WithText("invalid json").
+					WithHeader("content-type", "application/json").
+					Expect().Status(http.StatusBadRequest)
+			})
+		})
+
+		Context("With missing mandatory fields", func() {
+			It("returns 400", func() {
+				ctx.SMWithOAuth.POST("/v1/visibilities").
+					WithJSON(postVisibilityRequest).
+					Expect().Status(http.StatusCreated)
+
+				for _, prop := range []string{"service_plan_id"} {
+					delete(postVisibilityRequest, prop)
+
+					ctx.SMWithOAuth.POST("/v1/visibilities").
+						WithJSON(postVisibilityRequest).
+						Expect().Status(http.StatusBadRequest).JSON().Object().Keys().Contains("error", "description")
+				}
+			})
+		})
+
+		Context("with not existing related platform", func() {
+			It("returns 400", func() {
+				platformId := "not-existing"
+				ctx.SMWithOAuth.GET("/v1/platforms/"+platformId).
+					WithJSON(postVisibilityRequest).
+					Expect().Status(http.StatusNotFound).JSON().Object().Keys().Contains("error", "description")
+
+				ctx.SMWithOAuth.POST("/v1/visibilities").
+					WithJSON(common.Object{
+						"service_plan_id": existingPlanIDs[0],
+						"platform_id":     platformId,
+					}).
+					Expect().Status(http.StatusBadRequest).JSON().Object().Keys().Contains("error", "description")
+			})
+		})
+
+		Context("with missing platform id field", func() {
+			It("returns 201 if no visibilities for the plan exist", func() {
+				ctx.SMWithOAuth.GET("/v1/visibilities").
+					Expect().Status(http.StatusOK).JSON().Path("$.visibilities[*].id").Array().NotContains(existingPlanIDs[1])
+
+				ctx.SMWithOAuth.POST("/v1/visibilities").
+					WithJSON(common.Object{
+						"service_plan_id": existingPlanIDs[1],
+					}).
+					Expect().Status(http.StatusCreated).JSON().Object().ContainsMap(common.Object{
+					"service_plan_id": existingPlanIDs[1],
+				})
+			})
+
+			It("returns 400 if visibilities for the plan exist", func() {
```

Comment:
how do you make a plan public then? DELETE request for N visibilities with N platform_ids and service_plan and then a new POST?

Reply:
you would need to clean up all visibilities related to the plan before you can create a public visibility.

I am up for alternative suggestions as well.

Reply:
Should be noted in specification